File: src/backend/TrainingManagement.Api/Entities/Employee.cs
using System;

namespace TrainingManagement.Api.Entities;

public sealed class Employee
{
	// 员工编号，主键
    public long EmpId { get; set; }

	// 登录名
    public string LoginName { get; set; } = string.Empty;

    // 密码哈希
    public string PasswordHash { get; set; } = string.Empty;

    // 员工姓名
    public string EmpName { get; set; } = string.Empty;

    // 部门名称，外键，关联DepartmentsTraining.DeptName
    public string DeptName { get; set; } = string.Empty;

    // 职位
    public string? Position { get; set; }

    // 邮箱
    public string? Email { get; set; }

    // 电话
    public string? Phone { get; set; }

    // 入职日期
    public DateTime HireDate { get; set; }

    // 员工状态
    public string Status { get; set; } = "ACTIVE";

    // 创建日期
    public DateTime CreatedAt { get; set; }
}

File: src/frontend/src/api/mappers/employee.ts
import { fromEnvelopeFailure } from '@/api/error'
import type { ApiEnvelopeDto, EmployeeSummaryDto, PageDto } from '@/api/transport'
import type { PageResult } from '@/types/api'
import type { EmployeeSummary } from '@/domains/employee'

export function mapEmployeeSummary(dto: EmployeeSummaryDto): EmployeeSummary {
  return {
    empId: dto.empId,
    empName: dto.empName,
    deptName: dto.deptName,
    position: dto.position || '—',
    email: dto.email ?? null,
    phone: dto.phone ?? null,
    hireDate: dto.hireDate ?? null,
    status: dto.status === 'RESIGNED' ? 'RESIGNED' : 'ACTIVE',
    createdAt: dto.createdAt ?? null,
  }
}

export function mapEmployee(envelope: ApiEnvelopeDto<EmployeeSummaryDto>): EmployeeSummary {
  if (!envelope.success || !envelope.data) throw fromEnvelopeFailure(envelope)
  return mapEmployeeSummary(envelope.data)
}

export function mapEmployeePage(
  envelope: ApiEnvelopeDto<PageDto<EmployeeSummaryDto>>,
): PageResult<EmployeeSummary> {
  if (!envelope.success || !envelope.data) throw fromEnvelopeFailure(envelope)
  return { ...envelope.data, items: envelope.data.items.map(mapEmployeeSummary) }
}
